### DISCUSSION TITLE
WebFlux support in spring dgs-framework

### DISCUSSION THREAD

**Discussion by Nokinori** [Netflix/dgs-framework#294] [Ideas]
Hi there!
First of all, I wanted to ask about current progress in webflux support? I can't find any created discussion yet.
Or mb you need some help with that feature? 

And about implementation. How it will be done? 
With Project Reactor?
`val executionResult: Mono<ExecutionResult> = Mono.fromCompletionStage(graphQL.executeAsync(invocationInput))`
Or Coroutines with suspend functions?
`val executionResult: ExecutionResult graphQL.executeAsync(invocationInput).await()`

Also, what is about whole coroutine support? Like suspend functions for `@DgsData` and other stuff.
Or it's not the main goal of this task?

About internal implementation.
`com.netflix.graphql.dgs.DgsQueryExecutor` provide only synchronous `execute` and `executeAndExtractJsonPath` methods. So there will be new `***Async` or a new interface like `DgsQueryExecutorAsync`?

I just want to discus the idea, because there are different ways of implementation )))

**Comment by paulbakker** [comment]
Hi @Nokinori, I have a work-in-progress branch as of last week with the basics working: https://github.com/netflix/dgs-framework/tree/webflux. Query execution works, and there's a WebFlux powered HTTP endpoint, but there are still features missing such as GraphiQL and file uploads. I'm hoping to finish most of that this week.

The idea is to use Reactor. I've added a new interface [DgsReactiveQueryExecutor](https://github.com/Netflix/dgs-framework/blob/webflux/graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsReactiveQueryExecutor.java), which is basically the same as the existing `DgsQueryExecutor`, but using `Mono`. 
The interface is implemented by using the `executeAsync` from `graphql-java` internally.

I've not gone the coroutines way because I don't think it's the best way forward for Java users, although it would be nice for those using Kotlin. Maybe this is something for later, but out of scope for now.

**Reply by Nokinori** [reply]
Hi @paulbakker, thank you for answer. 
Looked through the branch. You've done a great job there!
I have a few questions, but I will wait for the pr)

**Reply by paulbakker** [reply]
PR ready for review: https://github.com/Netflix/dgs-framework/pull/301

**Comment by ahsaid** [comment]
@paulbakker do you support  spring webflux now? I don't see any documentation yet in https://netflix.github.io/ ?

**Reply by paulbakker** [reply]
Yes, that added a few months ago. The only change is using a different dependency: https://netflix.github.io/dgs/getting-started/#adding-the-dgs-framework-dependency. 
The programming model is exactly the same, but it changes the underlying implementation.
